plugins/camera/camera.h

```diff
@@ -0,0 +1,381 @@
+        constexpr static const uint16_t RESOLUTION_H_HIGHEST = UINT16_MAX;
+        constexpr static const uint16_t RESOLUTION_V_HIGHEST = UINT16_MAX;
+        constexpr static const uint32_t BIT_RATE_AUTO = UINT32_MAX;
+    };
+
+    /**
+     * @brief Sets video stream settings.
+     *
+     * @param settings Reference to custom video stream settings which include resolution,
+     * framerate, bitrate, etc.
+     */
+    void set_video_stream_settings(const VideoStreamSettings &settings);
+
+    /**
+     * @brief Information about video stream.
+     */
+    struct VideoStreamInfo {
+        VideoStreamSettings settings; /**< @brief Video stream settings. */
+        enum class Status {
+            NOT_RUNNING = 0, IN_PROGRESS
+        } status; /**< @brief Current status of video streaming. */
+    };
+
+    /**
+     * @brief Get video stream information (synchronous).
+     *
+     * Application may use media player like VLC and feed `uri`
+     * to play the ongoing video streaming.
+     *
+     * @param info Video stream information will be filled.
+     *
+     * @return SUCCESS if video stream info is received, error otherwise.
+     */
+    Result get_video_stream_info(VideoStreamInfo &info);
+
+    /**
+     * @brief Starts video streaming (synchronous).
+     *
+     * Sends a request to start video streaming.
+     *
+     * @sa stop_video_streaming()
+     *
+     * @return SUCCESS if video streaming is started, error otherwise.
+     */
+    Result start_video_streaming();
+
+    /**
+     * @brief Stop the current video streaming (synchronous).
+     *
+     * Sends a request to stop ongoing video streaming.
+     *
+     * @sa start_video_streaming()
+     *
+     * @return SUCCESS if video streaming is stopped, error otherwise.
+     */
+    Result stop_video_streaming();
+
+    /**
+     * @brief Callback type for capture info updates.
+     */
+    typedef std::function<void(CaptureInfo)> capture_info_callback_t;
+
+    /**
+     * @brief Subscribe to capture info updates (asynchronous).
+     *
+     * @param callback Function to call with updates.
+     */
+    void capture_info_async(capture_info_callback_t callback);
+
+    /**
+     * @brief Information about camera status.
+     */
+    struct Status {
+        bool video_on;
+        bool photo_interval_on;
+
+        enum class StorageStatus {
+            NOT_AVAILABLE,
+            UNFORMATTED,
+            FORMATTED
+        } storage_status;
+        float used_storage_mib;
+        float available_storage_mib;
+        float total_storage_mib;
+    };
+
+    /**
+     * @brief Callback type to get status.
+     */
+    typedef std::function<void(Result, const Status &)> get_status_callback_t;
+
+
+    /**
+     * @brief Get camera status.
+     *
+     * @param callback Function to call with camera status.
+     */
+    void get_status_async(get_status_callback_t callback);
+
+    bool get_possible_settings(std::vector<std::string> &settings);
```

Comment:
Are docs needed for this and following. Ideally everything in public API should be documented.

Reply:
Will do.